### ISSUE TITLE
net: isDomainName should be exported

### ISSUE THREAD

**Issue by chet** [golang/go#31671]
I'd like to essentially re-open https://github.com/golang/go/issues/2880 for consideration.

The motivating factor here is I have yet to find a good canonical piece of code which callers can leverage to validate an input name against [RFC 1035](https://tools.ietf.org/html/rfc1035). Searching around for such code yields a whole world of regexes people have come up with that don't fully implement the spec. `net.isDomainName` may not be perfect either (see: https://github.com/golang/go/issues/17659), but it would do a better job that what I'm finding out there otherwise (and be faster).

The original issue got kicked around a lot and eventually closed, so I'm hoping for some feedback one way or the other this time around.

Thanks!

**Comment by bradfitz** [comment]
We definitely don't want to do this while all the questions from #17659 remain unanswered. Otherwise how would we document this function?

**Comment by chet** [comment]
@bradfitz Definitely. My main hope is a decision to this proposal can be made, one way or another. If a decision is made in favor of the proposal, I agree this needs to remain open until #17659 is resolved, and then go from there.

**Comment by rsc** [comment]
Putting this proposal on hold until at least #17659 is resolved.
